cornflow-server/cornflow/shared/authentication/auth.py

```diff
@@ -195,65 +187,175 @@ def get_token_from_header(headers: Headers = None) -> str:
+                log_txt=f"Error while trying to get a token from header. " + err,
+                status_code=400
+            )
+
         try:
-            return auth_header.split(" ")[1]
+            token = auth_header.split(" ")[1]
+            return token
         except Exception as e:
-            err = f"The authorization header has a bad syntax: {e}"
+            err = "Invalid Authorization header format. Must be 'Bearer <token>'"
             raise InvalidCredentials(
-                err, log_txt=f"Error while trying to get a token from header. " + err
+                err, 
+                log_txt=f"Error while trying to get a token from header. " + err,
+                status_code=400
             )
 
     def get_user_from_header(self, headers: Headers = None) -> UserModel:
         """
-        Gets the user represented by the token that has to be in the request headers.
+        Extracts the user from the Authorization headers.
 
         :param headers: the request headers
         :type headers: `Headers`
         :return: the user object
-        :rtype: `UserBaseModel`
+        :rtype: :class:`UserModel`
         """
         if headers is None:
-            err = "Headers are missing from the request. Authentication was not possible to perform."
+            err = "Request headers are missing"
             raise InvalidUsage(
-                err, log_txt="Error while trying to get user from header. " + err
+                err, 
+                log_txt="Error while trying to get user from header. " + err,
+                status_code=400
             )
         token = self.get_token_from_header(headers)
         data = self.decode_token(token)
-        user_id = data["user_id"]
-        user = self.user_model.get_one_user(user_id)
+        
+        user = self.user_model.get_one_object(username=data["username"])
+            
         if user is None:
-            err = "User does not exist, invalid token."
-            raise ObjectDoesNotExist(
-                err, log_txt="Error while trying to get user from header. " + err
+            err = "User not found. Please ensure you are using valid credentials"
+            raise InvalidCredentials(
+                err, 
+                log_txt="Error while trying to get user from header. User does not exist.",
+                status_code=400
             )
         return user
 
     @staticmethod
-    def return_user_from_token(token):
+    def get_public_keys(provider_url: str) -> dict:
         """
-        Function used for internal testing. Given a token gives back the user_id encoded in it.
+        Gets and caches the public keys from the OIDC provider
+        
+        :param str provider_url: The base URL of the OIDC provider
+        :return: Dictionary of kid to public key mappings
+        :rtype: dict
+        """
+        # Check cache first
+        if provider_url in public_keys_cache:
+            return public_keys_cache[provider_url]
```

Comment:
Actually this is not what you want to check that is on the cache, you want to check if the `kid` is in the cache. If it is not, then you read the url again and if you do not have it on cache and you do not get it back, then the `kid` is not valid.

Right know a newly generated kid if you do not have it on cache you would not ask again for it and authentication would fail when it does not have to